### DISCUSSION TITLE
golang not found when running go mod tidy

### DISCUSSION THREAD

**Discussion by monofone** [renovatebot/docker-renovate#1745] [Q&A] [answered]
When running renovate with gomodTidy or a postUpdateTasks of "go mod download", for the slim image one gets an error

```
           "lockFile": "go.mod",
           "stderr": "Command failed: go mod download\n/bin/sh: 1: go: not found\n"
```

Configuration is:

```
module.exports = {
  platform: "gitlab",
  enabledManagers: [
    "....",
    "gomod",
    "...",
  ],
  postUpdateOptions: [
    "gomodTidy",
  ], 
  // discovers every repository the user "renovate" has access to
  autodiscover: false,

  binarySource: "global",

  persistRepoData: true,
  requireConfig: "required",
  onboarding: true,
  onboardingConfig: {
    $schema: "https://docs.renovatebot.com/renovate-schema.json",
    extends: ["local>renovate/renovate-bot"],
  },
  allowedPostUpgradeCommands: ["^npx", "^go"],
};
``` 
And a repository configuration of:

``` 
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "local>renovate/renovate-bot"
  ],
  "prConcurrentLimit": 5,
  "constraints": {
    "go": "1.19"
  },
  "packageRules": [
    {
      "matchPackageNames": ["golang"],
      "rangeStrategy": "pin"
    }
  ],
  "postUpgradeTasks":
  {
    "commands": ["go mod download"]
  }
}
``` 
When skipping the `go mod download` command it complains about `go mod tidy` with the same message that `go` is not available in the path.

Any hints on configuration mistakes ? or is this a problem with the slim image ? the Full image is working as expected but way bigger.

**Comment by ga-andrew-b** [comment]
We get the same issue (using the `renovate/renovate:36` image).  The error we get is:
 ```txt
 "err": {
         "cmd": "/bin/sh -c install-tool golang 1.21.0",
         "stderr": "[11:04:59.098] INFO (407): Downloading file ...\n    url: \"https://dl.google.com/go/go1.21.linux-amd64.tar.gz\"
```
It looks like it is trying to download the wrong filename.  With the latest version of Go (Go 1.21 and onwards) it now includes a `.0` patch version in the name for the very first release (all prior releases of Go dropped the patch version number when it was a minor release), so it should be trying to download `https://dl.google.com/go/go1.21.0.linux-amd64.tar.gz` (see 'Introduction to Go 1.21' on the [release notes](https://go.dev/doc/go1.21)).

**Reply by viceice** [reply]
you've a different issue, please ensure to have the latest renovate image.

**Comment by viceice** [comment]
@monofone its expected to not have go available on post upgrade tasks. please ask ask on main renovate repo discussion for more help.

this repo is just for docker image issues

**Comment by monofone** [comment]
HI @viceice,

as stated in the original message this also happens for `go mod tidy` configured via `postUpdateOptions`. So for me this seems to be an image issue that go is not downloaded when needed and if it is downloaded it should be available for postUpgradeTasks too. But if `gomodTidy` works as expected I probably don't need it anyway.

**Reply by viceice** [reply, marked as answer]
no, it highly depends on how you're running renovate. go isn't always globally available.

so please open a discussion on main renovate repo for further help. I'll lock this because it's not an issue of the docker image
